packages/core/src/typeschema/validation.ts

```diff
@@ -589,7 +594,19 @@ export function matchDiscriminant(
       if (!value || !sliceElement) {
         return false;
       }
-      if (matchesSpecifiedValue(value, sliceElement)) {
+      if (sliceElement.pattern /*&& deepIncludes(value, sliceElement.pattern)*/) {
```

Comment:
This is a subtle but potentially breaking validation change for malformed slice definitions. Since `matchesSpecifiedValue` returns `true` by default, checks on elements not containing either `.fixed` nor `.pattern` now return `false` whereas they used to return `true`. The only time fixed and pattern shouldn't be defined is when we're dealing with slicing by ValueSet which is handled below and retains the permissiveness for now.